lib/ts/recipe/oauth2client/index.ts

```diff
@@ -13,20 +13,25 @@
  * under the License.
  */
 
-import { UserContext } from "../../types";
+import { getUserContext } from "../../utils";
 import Recipe from "./recipe";
 import { RecipeInterface, APIInterface, APIOptions, OAuthTokens } from "./types";
 
 export default class Wrapper {
     static init = Recipe.init;
 
-    static async getAuthorisationRedirectURL(redirectURIOnProviderDashboard: string, userContext: UserContext) {
+    static async getAuthorisationRedirectURL(
+        redirectURIOnProviderDashboard: string,
+        userContext?: Record<string, any>
+    ) {
         const recipeInterfaceImpl = Recipe.getInstanceOrThrowError().recipeInterfaceImpl;
-        const providerConfig = await recipeInterfaceImpl.getProviderConfig({ userContext });
+        const providerConfig = await recipeInterfaceImpl.getProviderConfig({
```

Comment:
these should be getting the same context object, so please extract `getUserContext(userContext)` to a constant and pass the same object to both recipe function calls.

Reply:
Fixed.